content/blog/2022/2022-06-30-configuring-cors-with-spring.md

```diff
@@ -0,0 +1,430 @@
+                    .allowedOrigins(cors.getAllowedOrigins())
+                    .allowedMethods(cors.getAllowedMethods())
+                    .maxAge(cors.getMaxAge())
+                    .allowedHeaders(cors.getAllowedHeaders())
+                    .exposedHeaders(cors.getExposedHeaders());
+            }
+        };
+    }
+````
+
+{{% info title="NOTE:" %}}
+**addMapping()** returns a **CorsRegistration** object which applies default **CorsConfiguration** if
+one or more methods(**allowedOrigins**, **allowedMethods**, **maxAge**, **allowedHeaders**, **exposedHeaders**) are not explicitly defined.
+Refer to the Spring library method **CorsConfiguration.applyPermitDefaultValues()** to understand the defaults applied.
+{{% /info %}}
+
+## Configuring CORS in a Spring Webflux application
+
+The initial setup created with a Spring Initializr uses Spring webflux, Spring Data R2DBC and H2 Database.
+No external dependencies need to be added.
+
+
+### CORS Configuration for Spring Webflux using @CrossOrigin
+
+Similar to Spring REST we can define **@CrossOrigin at the class level or at the method level** in case of Spring Webflux.
+The same @CrossOrigin attributes described in the previous sections will apply. Also, when the annotation is defined at both class and method,
+its combined attributes will apply to the methods.
+
+````java
+    @CrossOrigin(origins = "http://localhost:4200", allowedHeaders = "Requestor-Type", exposedHeaders = "X-Get-Header")
+    @GetMapping
+    public ResponseEntity<Mono<List<BookDto>>> getBooks(@RequestParam String type) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("X-Get-Header", "ExampleHeader");
+        return ResponseEntity.ok().headers(headers).body(libraryService.getAllBooks(type));
+    }
+````
+
+#### Enable CORS Configuration globally in Spring Webflux
+
+To define CORS globally in a Spring Webflux application, we use the **WebfluxConfigurer** and override the **addCorsMappings()**.
+Similar to Spring REST, it uses a **CorsConfiguration** with defaults that can be overridden as required.
+
+````java
+    @Bean
+    public WebFluxConfigurer corsMappingConfigurer() {
+            return new WebFluxConfigurer() {
+        @Override
+        public void addCorsMappings(CorsRegistry registry) {
+            WebConfigProperties.Cors cors = webConfigProperties.getCors();
+            registry.addMapping("/**")
+            .allowedOrigins(cors.getAllowedOrigins())
+            .allowedMethods(cors.getAllowedMethods())
+            .maxAge(cors.getMaxAge())
+            .allowedHeaders(cors.getAllowedHeaders())
+            .exposedHeaders(cors.getExposedHeaders());
+            }
+        };
+    }
+````
+
+{{% info title="NOTE:" %}}
+In all the above cases we can define both global CORS configuration and local configuration using @CrossOrigin.
+For attributes that accept multiple values, a combination of global and local values will apply. For attributes that accept
```

Comment:
Aha, here's the information I wanted to know in my comment above. Please move this information to the place above then there's no need for this note here anymore.